### PR TITLE
Clarify image pull secret documentation

### PR DESCRIPTION
**Pull request by Jamstah**
### Description

The statement about missing pull secrets is incorrect, the kubelet will still attempt to pull the image if a pull secret is missing.

Added some information about how multiple pull secrets are handled.

### Issue

Closes: n/a

### CODE REVIEW

**Comment by Jamstah** [comment]
Shout out to https://medium.com/pareture/how-using-multiple-k8s-imagepullsecrets-works-77d2a2ca7a47

File: content/en/docs/tasks/configure-pod-container/pull-image-private-registry.md [hunk 1]
```diff
@@ -238,6 +236,14 @@ Events:
   ...  FailedToRetrieveImagePullSecret  ...  Unable to retrieve some image pull secrets (<regcred>); attempting to pull the image may not succeed.
 ```
 
+## Using images from multiple registries
+
+A pod can have multiple containers, each container image can be from a different registry.
+You can use multiple pull secrets with one pod, and pull secrets can contain multiple credentials.
```
Comment (salaxander) [inline content/en/docs/tasks/configure-pod-container/pull-image-private-registry.md:242]:
```suggestion
You can use multiple `imagePullSecrets` with one pod, and each can contain multiple credentials.
```
given that `imagePullSecrets` is an object field, I think according to the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-object-field-names-and-namespaces), we should code style and camel case that.

File: content/en/docs/tasks/configure-pod-container/pull-image-private-registry.md [hunk 2]
```diff
@@ -238,6 +236,14 @@ Events:
   ...  FailedToRetrieveImagePullSecret  ...  Unable to retrieve some image pull secrets (<regcred>); attempting to pull the image may not succeed.
 ```
 
+## Using images from multiple registries
+
+A pod can have multiple containers, each container image can be from a different registry.
+You can use multiple `imagePullSecrets` with one pod, and each can contain multiple credentials.
+
+The image pull will be attempted using each credential that matches the registry.
+If no credentials match the registry, the image pull will be attempted without authorization.
```
Comment (saschagrunert) [inline content/en/docs/tasks/configure-pod-container/pull-image-private-registry.md:245]: Technically runtimes can be configured to auth against registries without Kubernetes being aware of it. I think mentioning it should be sufficient, like:

```suggestion
If no credentials match the registry, the image pull will be attempted without authorization or
custom runtime specific configuration.
```

Comment (Jamstah) [reply content/en/docs/tasks/configure-pod-container/pull-image-private-registry.md]: Added.

I was trying to stay away from the other mechanisms in this section to not confuse things, but I guess there are related links for people to look at for more detail.

**Review by saschagrunert — APPROVED** [review]

**Review by dipesh-rawat — COMMENTED** [review]
Thanks!

/lgtm

**Pull request merged**